### ISSUE TITLE
Background Local Evaluation Thread dies if temporary connection issue

### ISSUE THREAD

**Issue by therabidbanana** [amplitude/experiment-ruby-server#54]
We've noticed problems rolling out a new flag in the last 24 hours - the changes are not applying. We use Local Eval, and it appears the background thread has died.

## Expected Behavior

Temporary network issues should not cause servers to lose ability to update flag configuration

## Current Behavior

Any network failure during Flag fetch for local eval kills update thread and prevents further polling.

## Possible Solution

Error in my local machine that was running overnight:

```
#<Thread:0x0000000110e65cd8 /Users/david/.rvm/gems/ruby-3.1.3/gems/amplitude-experiment-1.2.2/lib/experiment/local/client.rb:95 run> terminated with exception (report_on_exception is true):
/Users/david/.rvm/rubies/ruby-3.1.3/lib/ruby/3.1.0/net/http.rb:1018:in `initialize': Failed to open TCP connection to api.lab.amplitude.com:443 (getaddrinfo: nodename nor servname provided, or not known) (SocketError)
```

I suspect errors should be caught and retried (with some safe backoff period, maybe just try again next cycle?) and the gem could probably test for this case.

## Steps to Reproduce

1. Set up local evaluation on Rails server
2. Disable network connection after booting Rails server
3. Wait until next Local eval update from https://github.com/amplitude/experiment-ruby-server/blob/main/lib/experiment/local/fetcher.rb#L27
4. See logs about dead thread due to exception. 
5. Confirm polling for flag updates has stopped even if connection reestablished

## Environment

- SDK Version: 1.2.2
- Language Version: Ruby 3.1.3

**Comment by therabidbanana** [comment]
If it's helpful, we can see this connectivity issue occurred during an hour period on Sept 26th around 5pm Pacific. Unclear if the connectivity issue is persistent but we've deployed one of the impact services several times since and not seen logs in last 24 hours. 

![Screenshot 2023-09-28 at 11 46 29 AM](https://github.com/amplitude/experiment-ruby-server/assets/18348/c8e99eeb-a1ae-4456-8784-9fcded959ad4)

**Comment by tyiuhc** [comment]
@therabidbanana, many thanks for raising this issue! It has been addressed in v1.2.4
